src_c/geometry.c

```diff
@@ -8,6 +8,108 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
+static PyObject *
+pg_raycast(PyObject *_null, PyObject *const *args, Py_ssize_t nargs)
+{
+    double start_x, start_y;
+    double end_x, end_y;
+    PyObject **farr;
+    Py_ssize_t col_length;
+    Py_ssize_t loop;
+    double angle;
+    double max_dist;
+
+    if (nargs == 3) {
+        if (!PySequence_Check(args[0]) || !PySequence_Check(args[1]) || !PySequence_FAST_CHECK(args[2])) {
```

Comment:
I feel these checks are superflous as `pg_TwoDoublesFromObj` ( the statement right after this one ) already checks and errors out if it couldn't retrieve the two values. You are basically doing twice the amount of necessary checks.
```suggestion
        if (!PySequence_FAST_CHECK(args[2])) {
```

Reply:
Remove these, it's not just my feeling, they're actually not needed.